src/attrib.d

```
@@ -611,23 +624,91 @@ public:
+        return new AlignDeclaration(loc,
+            ealign.syntaxCopy(), Dsymbol.arraySyntaxCopy(decl));
     }
 
     override Scope* newScope(Scope* sc)
     {
-        return createNewScope(sc, sc.stc, sc.linkage, sc.protection, sc.explicitProtection, this.salign, sc.inlining);
+        return createNewScope(sc, sc.stc, sc.linkage,
+            sc.protection, sc.explicitProtection, this,
+            sc.inlining);
+    }
+
+    override void setScope(Scope* sc)
+    {
+        //printf("AlignDeclaration::setScope() %p\n", this);
+        if (ealign && decl)
+            Dsymbol.setScope(sc); // for forward reference
+        return AttribDeclaration.setScope(sc);
+    }
+
+    override void semantic2(Scope* sc)
+    {
+        getAlignment();
+        super.semantic2(sc);
+    }
+
+    structalign_t getAlignment()
+    {
+        if (!ealign)
+            return STRUCTALIGN_DEFAULT;
+
+        if (auto sc = _scope)
+        {
+            _scope = null;
+
+            sc = sc.startCTFE();
+            ealign = ealign.semantic(sc);
+            ealign = resolveProperties(sc, ealign);
+            sc = sc.endCTFE();
+
+            auto showError()
+            {
+                .error(loc, "positive integer expected, not %s", ealign.toChars());
+                return STRUCTALIGN_DEFAULT;
+            }
+
+            if (ealign.op == TOKerror)
+                return STRUCTALIGN_DEFAULT;
+            if (!ealign.type)
+                return showError();
+            Type tb = ealign.type.toBasetype();
+            if (!tb.isintegral())
+                return showError();
+            if (tb.ty == Tchar || tb.ty == Twchar || tb.ty == Tdchar || tb.ty == Tbool)
+                return showError();
+
+            ealign = ealign.ctfeInterpret();
+            if (ealign.op == TOKerror)
+                return STRUCTALIGN_DEFAULT;
+
+            auto n = ealign.toInteger();
+            if (!tb.isunsigned() && n > long.max)
```

No, `n` is `dinteger_t`, it's unsigned.

And when `n` is bigger than `structalign_t.max`, currently it's silently accepted.
https://github.com/dlang/dmd/pull/5750/files/891ecfbf40c6a7a657c897cff9ea804eee5f5dd0#diff-e55efe6f4373bd0f2e2da9a75322a544L981

Should I add a new error check in here?